docs/pages/07_service.adoc

```diff
@@ -0,0 +1,955 @@
+
+The main building blocks of the service model are transfer objects, services, exports and actors. The following chapters describe these building blocks in detail.
+
+== Transfer object
+
+A transfer object is a container for a set of data fields that is transferred between the service layer and any upper layers (e.g. user interface or external system).
+
+[NOTE]
+====
+The term "transfer object" is a short version of "transfer object type". The correct term would be "transfer object type". However, to distinguish between the type and the instance, the object is called "instance of a transfer object".
+====
+
+There are two categories of transfer objects:
+
+* unmapped and
+* mapped transfer objects.
+
+// Additionally, transfer objects may contain activities such as internal consistency checking, validation and services.
+
+=== Unmapped transfer object
+
+Unmapped transfer objects are simple non-persistent objects that carry data between our application and the outside world. The data fields of an unmapped transfer object can be set and read at any stage of its life cycle.
+
+To define an unmapped transfer object, use the `transfer` keyword.
+
+*Syntax:*
+
+----
+transfer <name> {
+    [field] ...
+}
+----
+
+where the <name> is the name of the transfer object, and its data fields are defined between `{` and `}`. Possible field definitions are provided in the next section.
+
+*Example:*
+
+----
+transfer PersonTransfer {
+}
+----
+
+The example above defines an empty transfer object named PersonTransfer. This transfer object has no fields.
+
+=== Mapped transfer object
+
+Data transformation between entities and transfer objects is a tedious and error-prone task. The mapped transfer objects are intended to facilitate this work. Mapped transfer objects also carry data between our application and the outside world, but they can be linked to a specific entity instance. The link between the mapped transfer object and the entity is represented by a specific field of the mapped transfer object, called mapping field.
+
+The mapping is a configuration approach that copies entity field values into the fields of the transfer object in a pre-defined way.
+
+To define a mapped transfer object, use the `transfer` keyword.
+
+*Syntax:*
+
+----
+transfer <name> (<entity> <mapping-field>) [{
+    [field] ...
+}]
+----
+
+where the <name> is the name of the transfer object, and the members are defined between `{` and `}`. The <entity> is the entity type to which the transfer object is linked and <mapping-field> is the name of the field that can be referenced. Possible field definitions are provided in the next section.
```

Comment:
I would replace the <mapping-field> term/name to the <mapping-parameter> or <parameter-name>. Because when I first read it. I associated that it is a field of the entity. And the used syntax and the meaning of it is more like a parameter than a field.

Reply:
I think when keyword is used it is a better and easier to understand:

`transfer <name> mapped to <entity> [as <mapping-field>]`

When alias is not used `self` keyword can be used. When mapping-field is defined it can be used as reference.

Reply:
I don't agree with the `self` keyword in this context, because the `self` represent the transfer object instance in the other context in the JUDO (for example derived fields). And the self keyword means the direct context in every other language (for example: OCL, Perl, ...). Therefore I used a new keyword for this. For example: `linkedObject`, `mappedObject`, `mappedInstance`, `linkedInstance` or any similar and term that meaning is very clear and unambiguous.

Mendix has introduced the `$currentObject` keyword in similar context. Exactly, it is a predefined variable, of which the type and value depends on the context.